common/patches/rof2.cpp

```diff
@@ -1582,27 +1583,86 @@ namespace RoF2
 		uchar* __emu_buffer = in->pBuffer;
 
 		ItemPacket_Struct* old_item_pkt = (ItemPacket_Struct*)__emu_buffer;
-		EQ::InternalSerializedItem_Struct* int_struct = (EQ::InternalSerializedItem_Struct*)(&__emu_buffer[4]);
 
-		EQ::OutBuffer ob;
-		EQ::OutBuffer::pos_type last_pos = ob.tellp();
+		switch(old_item_pkt->PacketType) 			
+		{
+			case ItemPacketParcel: {
+				ParcelMessaging_Struct       pms{};
+				EQ::Util::MemoryStreamReader ss(reinterpret_cast<char *>(in->pBuffer), in->size);
+				cereal::BinaryInputArchive   ar(ss);
+				ar(pms);
+
+				uint32 serial_length      = pms.serialized_item.length();
+				uint32 packet_type        = pms.packet_type;
+				uint32 date_sent          = pms.sent_time;
+				uint32 player_name_length = pms.player_name.length();
+				uint32 note_length        = pms.note.length();
+				char   *serial            = new char[serial_length];
+				char   *player_name       = new char[player_name_length];
+				char   *note              = new char[note_length];
+
+				strncpy(player_name, pms.player_name.c_str(), player_name_length);
+				strncpy(note, pms.note.c_str(), note_length);
+				strncpy(serial, pms.serialized_item.c_str(), serial_length);
```

Comment:
Are these char buffers/copies even needed? `serial` looks unused, can `player_name` and `note` just be written directly to the output buffer from the `pms` strings?

Reply:
Rewrote this to be more efficient.